chore: restructure folder layout for better organization

sanholl opened the pull request:
기존 강의 보고 진행했던 프로젝트를 FSD 아키텍처를 적용해서 구조 수정을 진행했습니다.

sanholl committed:
chore: restructure folder layout for better organization

dante01yoon commented:
Assignee가 아닌 Reviewers에 할당해주세요 :)

dante01yoon commented on todo-list/src/features/todo/context/TodoProvider.tsx:
```diff
@@ -1,7 +1,7 @@
+import { loadTodos } from "../storage/todoStorage";
 
 interface TodoProviderProps {
   children: ReactNode;
```
`ChildrenWithProps` 타입을 사용하여 children prop을 명시적으로 선언하지 않을 수 있습니다.

dante01yoon replied:
```ts
import { FC, ChildrenWithProps } from "react"

export const TodoProvider: FC<ChildrenWithProps<TodoProviderProps>> = ({ children }) => {
  .. 

```

dante01yoon commented on todo-list/src/features/todo/ui/TodoInput/index.tsx:
```diff
@@ -1,33 +1,12 @@
 import styles from "./TodoInput.module.css";
 import { ChangeEvent, FormEvent } from "react";
-import { useInputTodoDispatch, useInputTodoState, useTodoDispatch } from "../Todo/TodoProvider";
+import { useInputTodoDispatch, useInputTodoState, useTodoDispatch } from "../../context/TodoProvider";
```
`use*` 와 같은 훅은 별도 hooks 폴더 내부에 만드시는 것을 추천합니다.

dante01yoon commented on todo-list/src/entities/ui/index.ts:
```diff
@@ -0,0 +1 @@
+export * from "../../features/todo/ui/TodoItem";
```
entities 레이어는 features보다 하위 레이어이기 때문에 feature에서는 entities를 참조할 수 없습니다 :)

dante01yoon replied:
![스크린샷 2024-06-02 오전 9 07 59](https://github.com/sanholl/FE-sanhol/assets/6510430/d8a4c7c6-04dc-4b54-b0b8-397af4d53349)
https://feature-sliced.design/docs/get-started/overview

dante01yoon reviewed (commented):
고생하셨습니다. 이번 작업을 진행하시며 여러 고민을 하신 것이 느껴집니다. 👍  궁금하신 부분을 정리하셔서 온라인에서 뵙겠습니다.

sanholl committed:
feat: 코딩 테스트 풀이

sanholl committed:
fix: 구조 오류 수정
entities에서 feature폴더로 파일을 이동하면서 기존의 파일을 삭제하지 않고 entities에서 feature를 import 하는 문제가 생겨 해당 부분 삭제 및 수정 진행
